### PR TITLE
[develop] Pull in relevant doc updates from release v2.1.0

### LINKED ISSUE #497: Pull in relevant doc updates from release v2.1.0
## Description
The documentation was thoroughly revised for the v2.1.0 release, and some of those updates are relevant to the `develop` branch and need to be added. 

## Solution
Update the `develop` branch documentation.

### PR DESCRIPTION
**Pull request by gspetro-NOAA**
## DESCRIPTION OF CHANGES: 

The documentation was thoroughly revised for the v2.1.0 release, and some of those updates are relevant to the `develop` branch. This PR adds those updates to develop. 

### Type of change
- [X] This change requires a documentation update (all documentation)

## TESTS CONDUCTED: 
None required. Reviewed the docs on my fork's Read the Docs site: https://srw-ug.readthedocs.io/en/text-updates/

## DEPENDENCIES:
N/A.

## DOCUMENTATION:
All documentation.

## ISSUE: 
Fixes issue #497 .

## CHECKLIST
- [ ] My code follows the style guidelines in the Contributor's Guide
- [ ] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas N/A
- [X] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain). N/A
- [X] My changes generate no new warnings
- [ ] New and existing tests pass with my changes (N/A - changes are text only)
- [ ] Any dependent changes have been merged and published N/A

### CODE REVIEW

File: docs/UsersGuide/source/BuildSRW.rst
Comment (mark-a-potts) [inline docs/UsersGuide/source/BuildSRW.rst]: While this is technically true, it would be pretty unusual to clone into /home and would generally require sudo privileges to do so. I think the $HOME wording is better.

**Review by JeffBeck-NOAA — APPROVED** [review]

**Review by mark-a-potts — APPROVED** [review]

File: ush/Python/plot_allvars.py
Comment (MichaelLueken) [inline ush/Python/plot_allvars.py]: Is there a reason that you are reverting this change to `plot_allvars.py`?  The original changing from `Geopotential Height` to `Geopotential height` was due to a change in the naming of this field in `grib2`.  This naming change was causing the plotting scripts to fail.  Is this no longer an issue?

File: ush/Python/plot_allvars_diff.py
Comment (MichaelLueken) [inline ush/Python/plot_allvars_diff.py]: Is there a reason that you are reverting this change to `plot_allvars_diff.py`?  The original changing from `Geopotential Height` to `Geopotential height` was due to a change in the naming of this field in `grib2`.  This naming change was causing the plotting scripts to fail.  Is this no longer an issue?

**Review by MichaelLueken — COMMENTED** [review]
@gspetro-NOAA I have some comments related to the changes to the plotting scripts, but otherwise, the documentation changes look good to me.

For the modification to the plotting scripts, was it decided that these changes weren't necessary?  It sounded like a mismatch between `grib2` field naming for `Geopotential Height` was causing the scripts to fail (leading to the changing of the name originally).  If this is no longer an issue, then I'm fine with changing `Geopotential height` back to `Geopotential Height`.

File: ush/Python/plot_allvars.py
Comment (gspetro-NOAA) [reply ush/Python/plot_allvars.py]: @MichaelLueken Thanks for catching that! Not sure how that reversion slipped in there, but I have made `height` lowercase in both files now.

File: ush/Python/plot_allvars_diff.py
Comment (gspetro-NOAA) [reply ush/Python/plot_allvars_diff.py]: @MichaelLueken Thanks for catching that! Not sure how that reversion slipped in there, but I have made `height` lowercase in both files now.

**Review by MichaelLueken — APPROVED** [review]
@gspetro-NOAA Thanks for making the modifications to the plotting scripts!  Approving the PR now.

Is this work ready to be merged, or would you like for more eyes to look over it before it is merged?

**Comment by gspetro-NOAA** [comment]
> @gspetro-NOAA Thanks for making the modifications to the plotting scripts! Approving the PR now.
> 
> Is this work ready to be merged, or would you like for more eyes to look over it before it is merged?

@natalie-perlin wanted to take a look, so she's about to do that right now.

**Comment by MichaelLueken** [comment]
> > @gspetro-NOAA Thanks for making the modifications to the plotting scripts! Approving the PR now.
> > Is this work ready to be merged, or would you like for more eyes to look over it before it is merged?
> 
> @natalie-perlin wanted to take a look, so she's about to do that right now.

Thanks, @gspetro-NOAA!  I will wait for @natalie-perlin approval before merging this work.

File: docs/UsersGuide/source/RunSRW.rst
Comment (gspetro-NOAA) [inline docs/UsersGuide/source/RunSRW.rst:1181]: @MichaelLueken I just noticed that PR #475 changed PDY/cyc to PDYext/cycext. Should I update that here?

Comment (MichaelLueken) [reply docs/UsersGuide/source/RunSRW.rst:1181]: @gspetro-NOAA The change from `PDY/cyc` to `PDYext/cycext` was done in `scripts/exregional_get_extrn_mdl_files.sh` to correct an issue on WCOSS2 (the original fix was to directly alter the path of PDY and cyc, but that wasn't approved, leading to the creation of these two new variables).  The rest of the scripts still require PDY and cyc and these are are still being created in the `parm/FV3LAM_wflow.xml` XML file.

I think it is fine to keep the PDY and cyc as they currently are and not update them in the `RunSRW.rst` documentation.

**Review by natalie-perlin — CHANGES_REQUESTED** [review]
**Glossary.rst**, add the following to the end of line 121:

Updated documentation could be found at :doc:`HPC-Stack documentation <hpc-stack:index>`.

**Introduction.rst**, change in line 9 (partial):

... New and improved capabilities for the **v2.1.0** release included the addition of a verification package (METplus)...

**Quickstart.rst**, line 12 (partial) changes:

...  (2) configuring the stack to specify compilers, compiler options, and installation directory and (3) specifying the software packages and versions to be built, and building the stack that downloads necessary repositories or archives to build the requested packages. This process will create a number of modulefiles needed to build SRW App.

**Comment by MichaelLueken** [comment]
@natalie-perlin and @gspetro-NOAA For the modification to `Introduction.rst` line 9, would it be better to remove:

`New and improved capabilities for the v2.0.0 release included the addition of a verification package (METplus) for both deterministic and ensemble simulations and support for four stochastically perturbed physics schemes.`

This sentence appears to be a carry over from `release/public-v2`, so it is no longer necessary for `release/public-v2.1.0` and `develop`.

**Comment by gspetro-NOAA** [comment]
@natalie-perlin I made the Glossary update. Also added the Quickstart changes with slight rewording, but it seems that if users are just installing the default HPC-Stack, they wouldn't have to specify packages. I say "At a minimum..." so I want to be sure that specifying packages IS actually a required step. 
@MichaelLueken @natalie-perlin I would prefer to keep the Intro as is, since METplus and support for 4 physics suites (as opposed to 2) were significant improvements added in the v2.0.0 release and still present/applicable to the v2.1.0. I mentioned them in the past tense for users who are coming to this from the v1.x.x releases. Then switch to present tense for the current release's updates. I realize we're on `develop`, but since these changes reflect the most recent updates in `develop`, they give new users an idea of what the most recent progress/additions have included. I think when we get to a v3.0.0 release, we just replace that section with the new stuff. That was my thinking anyway.

**Comment by MichaelLueken** [comment]
> @MichaelLueken @natalie-perlin I would prefer to keep the Intro as is, since METplus and support for 4 physics suites (as opposed to 2) were significant improvements added in the v2.0.0 release and still present/applicable to the v2.1.0. I mentioned them in the past tense for users who are coming to this from the v1.x.x releases. Then switch to present tense for the current release's updates.

@gspetro-NOAA This makes sense to me.  Since this latest release is just a minor release, it is fine to keep the major updates from the version 2 major release in the introduction.  I'm fine with keeping the introduction as it currently is in this PR.

**Review by natalie-perlin — APPROVED** [review]

**Pull request merged**
